controllers/topics/index.js

```diff
@@ -170,105 +271,47 @@ const getTopics = async (req, res) => {
-
-    //Logic get topic
-    const topicService = new TopicService(Topics);
-    const topic = await topicService.getTopicByName(name);
-    const {topic_id} = topic;
-    const userTopicService = new UserTopicService(UserTopic, UserTopicHistory);
-
-    //get follow status
-    const [getTokenUserStatus, getSecondUserStatus] = await Promise.all([
-      userTopicService.getFollowTopicStatus(detailTokenUser.user_id, topic_id),
-      userTopicService.getFollowTopicStatus(secondDetailUser.user_id, topic_id)
-    ]);
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
 
-    let topicInvitationIds = [];
-    let data = [];
-    if (getTokenUserStatus) {
-      await userTopicService.unfollowTopic(detailTokenUser.user_id, topic_id);
-      data.push(
-        await _afterPutTopic(
-          true,
-          token,
-          detailTokenUser.user_id,
-          name,
-          detailTokenUser.is_anonymous,
-          topicInvitationIds,
-          prevUserToken,
-          with_system_message
-        )
-      );
-    } else {
-      //get invitations topic detail
-      const topicInvitations = await TopicInvitations.findAll({
-        where: {
-          user_id_invited: userId,
-          topic_id
-        }
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
       });
+    }
 
-      if (topicInvitations.length > 0) {
-        topicInvitationIds = topicInvitations.map(
-          (topicInvitation) => topicInvitation.topic_invitations_id
-        );
-        await TopicInvitations.destroy({
-          where: {
-            topic_invitations_id: topicInvitationIds
-          }
-        });
-      }
+    const topicId = await insertTopics([name]);
+    await followTopicV2Func(req, res);
 
-      await userTopicService.followTopic(detailTokenUser.user_id, topic_id);
-      data.push(
-        await _afterPutTopic(
-          false,
-          token,
-          detailTokenUser.user_id,
-          name,
-          detailTokenUser.is_anonymous,
-          topicInvitationIds,
-          prevUserToken,
-          with_system_message
-        )
-      );
-
-      if (getSecondUserStatus) {
-        topicInvitationIds = [];
-        await userTopicService.unfollowTopic(secondDetailUser.user_id, topic_id);
-        data.push(
-          await _afterPutTopic(
-            true,
-            token,
-            secondDetailUser.user_id,
-            name,
-            detailTokenUser.is_anonymous,
-            topicInvitationIds,
-            prevUserToken,
-            with_system_message
-          )
-        );
-      }
-    }
+    return res.status(200).json({
+      success: true,
+      message: 'Success',
+      topicId
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `createTopic` function, the error handling could be improved. Currently, it returns a JSON response with a status code and message, but it would be better to throw an error and let a global error handler take care of it. This way, you can ensure consistent error responses throughout your application.

```diff
- return res.status(400).json({
-   code: 400,
-   status: 'failed',
-   message: 'Topic already exists'
- });
+ throw new Error('Topic already exists');

- return res.json({
-   code: error.statusCode,
-   status: 'fail',
-   message: error.message,
-   data: 'null'
- });
+ throw error;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->